packages/flutter_tools/lib/src/flutter_cache.dart

```diff
@@ -139,9 +137,12 @@ class MaterialFonts extends CachedArtifact {
     ArtifactUpdater artifactUpdater,
     FileSystem fileSystem,
     OperatingSystemUtils operatingSystemUtils,
-  ) {
-    final Uri archiveUri = _toStorageUri(version);
-    return artifactUpdater.downloadZipArchive('Downloading Material fonts...', archiveUri, location);
+  ) async {
+    final String? cacheVersion = version;
+    if (cacheVersion != null) {
```

Comment:
Not sure if there's anything to be done if the version can't be parsed, it should always be present at this point.

Reply:
I think the right thing to do is `!` so if it does fail it shows up in crash logging

Reply:
Done, had to update an additional unit test.